### PR TITLE
Add scaffolding for e2e channel upgrade test

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #3421

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by damiannolan — APPROVED** [review]
Is the idea to merge this now and add to it incrementally? Happy to go with that approach if you'd like. 

I think it may be a nicer home for this test under `e2e/tests/core/04-channel` instead of `e2e/tests/upgrades` as all of the other tests under the upgrades directory are for coordinated chain upgrades/migrations.

**Review by DimitrisJim — APPROVED** [review]
lgtm too. I'd also support this going into a `core/04-channel` folder.

**Review by charleenfei — APPROVED** [review]

File: e2e/tests/core/02-client/client_test.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package e2e
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply e2e/tests/core/02-client/client_test.go]: Thanks, noticed that a while back and meant to change it!

**Comment by chatton** [comment]
converting to draft while fixing CI failures after moving client test files.

File: .github/workflows/e2e.yaml [hunk 2]
```diff
@@ -55,7 +55,11 @@ jobs:
             test_dirs="$(ls -A tests)"
             for td in $test_dirs
             do
-              go test -c "./tests/${td}"
+              # only test directories that contain go files.
+              if ls "./tests/${td}" | grep .go > /dev/null;
+              then
+                go test -c "./tests/${td}"
+              fi
```
Comment (chatton) [inline .github/workflows/e2e.yaml:62]: there are no go files in `e2e/tests/core` so this command was failing.

**Review by damiannolan — APPROVED** [review]
ack

**Pull request merged**